src/pybind/mgr/ansible/module.py

```diff
@@ -84,8 +78,113 @@ def result(self):
 
     @property
     def status(self):
-        """Return the status code of the operation
-        updating conceptually 'linked' attributes
+        """Retrieve the current status of the operation and update state
+        attributes
+        """
+        raise NotImplementedError()
+
+class ARSOperation(AnsibleReadOperation):
+    """Execute an Ansible Runner Service Operation
+    """
+
+    def __init__(self, client, logger, url, get_operation=True, payload=None):
+        """
+        :param client        : Ansible Runner Service Client
+        :param logger        : The object used to log messages
+        :param url           : The Ansible Runner Service URL that provides
+                               the operation
+        :param get_operation : True if operation is provided using an http GET
+        :param payload       : http request payload
+        """
+        super(ARSOperation, self).__init__(client, logger)
+
+        self.url = url
+        self.get_operation = get_operation
+        self.payload = payload
+
+    def __str__(self):
+
+            return "Ansible Runner Service: {operation} {url}".format(
+                operation = "GET" if self.get_operation else "POST",
+                url = self.url)
+
+    @property
+    def status(self):
+        """ Execute the Ansible Runner Service operation and update the status
+        and result of the underlying Completion object.
+        """
+
+        # Execute the right kind of http request
+        if self.get_operation:
+            response = self.ar_client.http_get(self.url)
+        else:
+            response = self.ar_client.http_post(self.url, self.payload)
+
+        # If no connection errors, the operation is complete
+        self._is_complete = True
+
+        # Depending of the response, status and result is updated
+        if not response:
+            self._is_errored = True
+            self._status = ExecutionStatusCode.ERROR
+            self._result = "Ansible Runner Service not Available"
+        else:
+            self._is_errored = (response == requests.codes.ok)
+
+            if not self._is_errored:
+                self._status == ExecutionStatusCode.SUCCESS
```

Comment:
Statement has no effect. Does that mean, `self._status` is never actually used? Thus `self.status` is never used?

Reply:
Good catch!... Thx! ..
It was working because object response is always different from requests.codes.ok!

self.status is used in the orchestrator "wait" method. Each time you query status, you update the real status of the operation in the Ansible Runner Service, in the case of a playbook it could take several seconds to finish an operation (asynchronous operation), in the case of an operation over the service (ARSOperation) the status method is responsible of doing the operation and return in the same moment the result (synchronous operation)

Reply:
Did you pushed your fix for this statement?

Reply:
Yes... the code snippet that we see here in "conversation" page does not match with the "real" code. You can see this fact if you the "files changed" page. I do not know why happens this ....